vmm/src/sigsys_handler.rs

```diff
@@ -57,18 +112,28 @@ mod tests {
             libc::SYS_munmap,
             libc::SYS_rt_sigprocmask,
             libc::SYS_rt_sigreturn,
+            libc::SYS_sched_getaffinity,
             libc::SYS_set_tid_address,
             libc::SYS_sigaltstack,
+            libc::SYS_write,
         ];
+
         assert!(setup_seccomp(SeccompLevel::Basic(REQUIRED_SYSCALLS)).is_ok());
+        let sys_idx = libc::SYS_getpid as usize;
+        assert_eq!(METRICS.seccomp.bad_syscalls[sys_idx].count(), 0);
 
         // Calls the blacklisted SYS_getpid.
         let _pid = process::id();
 
-        // The signal handler should let the program continue.
-        assert!(true);
+        assert!(cpu_count() > 0);
 
-        // The reason this test doesn't check the failure metrics as well is that the signal handler
-        // doesn't work right with kcov - possibly because the process is being pinned to 1 core.
+        // We expect cpu_count to be 1 when kcov is testing coverage. When this changes, we'll have
+        // to find another way to detect kcov, because it messes with our signal handler getting
+        // the SIGSYS, and the following assertions no longer hold.
+        if cpu_count() > 1 {
```

Comment:
This check doesn't look very reliable, but I tried for some time to find an alternative for checking whether the tests are run with cargo kcov. I am basically leaving this comment here maybe somebody else has a better idea on how to exclude the following asserts when running with kcov.